auth/src/main/java/com/firebase/ui/auth/provider/TwitterProvider.java

```diff
@@ -70,21 +72,41 @@ public void failure(TwitterException exception) {
         mCallbackObject.onFailure(new Bundle());
     }
 
-    private IdpResponse createIdpResponse(TwitterSession twitterSession) {
-        return new IdpResponse(
-                TwitterAuthProvider.PROVIDER_ID,
-                null,
-                twitterSession.getAuthToken().token,
-                twitterSession.getAuthToken().secret);
-    }
+    private static class EmailCallback extends Callback<String> {
+        private TwitterSession mTwitterSession;
+        private WeakReference<IdpCallback> mCallbackObject;
+
+        public EmailCallback(TwitterSession session, IdpCallback callbackObject) {
+            mTwitterSession = session;
+            mCallbackObject = new WeakReference<>(callbackObject);
+        }
+
+        @Override
+        public void success(Result<String> emailResult) {
+            mCallbackObject.get().onSuccess(createIdpResponse(emailResult.data));
```

Comment:
We need to check if `mCallbackObject.get()` is null here before calling `onSuccess` to avoid a NPE.

When it's `null` I guess we can just log a warning and do nothing, it means the activity waiting for the callback was dead so there's nowhere to deliver it.